embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/telemetry/EmbraceTelemetryService.kt

```diff
@@ -0,0 +1,77 @@
+package io.embrace.android.embracesdk.telemetry
+
+import io.embrace.android.embracesdk.capture.metadata.EmbraceMetadataService
+import io.embrace.android.embracesdk.internal.spans.toEmbraceAttributeName
+import java.util.Locale
+
+/*
+    Service for tracking usage of public APIs, and different internal metrics about the app.
+ */
+internal class EmbraceTelemetryService(
+    private val okHttpReflectionFacade: OkHttpReflectionFacade = OkHttpReflectionFacade()
+) {
+
+    private val usageCountMap = mutableMapOf<String, Int>()
+    private val appAttributesMap = mutableMapOf<String, String>()
+
+    /*
+        Tracks the usage of a public API by name. Adds a suffix for easier identification. Replaces whitespaces with underscores.
+    */
+    fun onPublicApiCalled(name: String) {
+        val suffixedName = "emb.usage.${
+        name.toLowerCase(Locale.ENGLISH)
+            .trim()
+            .replace(" ", "_")
+        }"
+        usageCountMap[suffixedName] = (usageCountMap[suffixedName] ?: 0) + 1
+    }
+
+    /*
+        Returns a map with every telemetry value. For now, it's just usage counts of public APIs.
+     */
+    fun getTelemetryAttributes(): Map<String, String> {
+        val telemetryMap = mutableMapOf<String, String>()
+
+        telemetryMap.putAll(getUsageCountTelemetry())
+
+        telemetryMap.putAll(getAppAttributes())
+
+        return telemetryMap
+    }
+
+    private fun getUsageCountTelemetry() = usageCountMap.mapValues {
+        it.value.toString()
+    }.also {
+        usageCountMap.clear()
```

Comment:
Is it necessary to reset this each time the function is called? Might we want to keep the counts for each span?

Reply:
We only add telemetry to the "current session" span. As getUsageCountTelemetry will only be called when the session ends, we need to flush the counts so we don't send the same ones for the next session